downloader/src/main/kotlin/vcs/Mercurial.kt

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (c) 2017 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.downloader.vcs
+
+import ch.frankel.slf4k.info
+import ch.frankel.slf4k.warn
+
+import com.here.ort.downloader.VersionControlSystem
+import com.here.ort.util.ProcessCapture
+import com.here.ort.util.log
+
+import java.io.File
+import java.io.IOException
+
+object Mercurial : VersionControlSystem() {
+
+    private val extensionSparse = "sparse = "
+    private val extensionLargeFiles = "largefiles = "
+
+    override fun getWorkingDirectory(vcsDirectory: File) =
+            object : WorkingDirectory(vcsDirectory) {
+                override fun isValid(): Boolean {
+                    val repositoryRoot = runMercurialCommand(workingDir, "root")
+                    return workingDir.path.contains(
```

Comment:
You're doing a String comparison on paths here, which *might* go wrong e.g. when running from Git Bash on Windows. Better use Kotlin's `File.startsWith()`, I suggest.